### ISSUE TITLE
Provision of attributes to optimizers

### ISSUE THREAD

**Issue by WDee** [dmlc/mxnet#1418]
i had a quick question on how to provide arguments to optimzers.

Does code like per below work? 

```
stytole_model = mx.model.FeedForward(ctx=devs,
        symbol             = network,
        num_epoch          = 40,
        optimizer          = mx.optimizer.Adam(),
        learning_rate      = 0.0001)

or will the learning rate argument be ignored, and one either has to enter like this:

stytole_model = mx.model.FeedForward(ctx=devs,
        symbol             = network,
        num_epoch          = 40,
        optimizer          = mx.optimizer.Adam(learning_rate      = 0.0001)

or like this?

stytole_model = mx.model.FeedForward(ctx=devs,
        symbol             = network,
        num_epoch          = 40,
        optimizer          =  "adam",
        learning_rate      = 0.0001)

Many thanks in advance, 

W

**Comment by piiswrong** [comment]
2&3 will work. 1 doesn't
